internal/ls/findallreferences.go

```diff
@@ -0,0 +1,1493 @@
+	case ast.KindNoSubstitutionTemplateLiteral, ast.KindStringLiteral:
+		return len(node.Text()) == len(searchSymbolName) && (isLiteralNameOfPropertyDeclarationOrIndexAccess(node) ||
+			isNameOfModuleDeclaration(node) ||
+			isExpressionOfExternalModuleImportEqualsDeclaration(node) ||
+			(ast.IsCallExpression(node.Parent) && ast.IsBindableObjectDefinePropertyCall(node.Parent) && node.Parent.Arguments()[1] == node) ||
+			ast.IsImportOrExportSpecifier(node.Parent))
+	case ast.KindNumericLiteral:
+		return isLiteralNameOfPropertyDeclarationOrIndexAccess(node) && len(node.Text()) == len(searchSymbolName)
+	case ast.KindDefaultKeyword:
+		return len("default") == len(searchSymbolName)
+	}
+	return false
+}
+
+func isForRenameWithPrefixAndSuffixText(options refOptions) bool {
+	return options.use == referenceUseRename && options.providePrefixAndSuffixTextForRename
+}
+
+func skipPastExportOrImportSpecifierOrUnion(symbol *ast.Symbol, node *ast.Node, checker *checker.Checker, useLocalSymbolForExportSpecifier bool) *ast.Symbol {
+	if node == nil {
+		return nil
+	}
+	parent := node.Parent
+	if parent.Kind == ast.KindExportSpecifier && useLocalSymbolForExportSpecifier {
+		return getLocalSymbolForExportSpecifier(node.AsIdentifier(), symbol, parent.AsExportSpecifier(), checker)
+	}
+	// If the symbol is declared as part of a declaration like `{ type: "a" } | { type: "b" }`, use the property on the union type to get more references.
+	return core.FirstNonNil(symbol.Declarations, func(decl *ast.Node) *ast.Symbol {
+		if decl.Parent == nil {
+			// Ignore UMD module and global merge
+			if symbol.Flags&ast.SymbolFlagsTransient != 0 {
+				return nil
+			}
+			// Assertions for GH#21814. We should be handling SourceFile symbols in `getReferencedSymbolsForModule` instead of getting here.
+			panic(fmt.Sprintf("Unexpected symbol at %s: %s", node.Kind.String(), symbol.Name))
+		}
+		if decl.Parent.Kind == ast.KindTypeLiteral && decl.Parent.Parent.Kind == ast.KindUnionType {
+			return checker.GetPropertyOfType(checker.GetTypeFromTypeNode(decl.Parent.Parent), symbol.Name)
+		}
+		return nil
+	})
+}
+
+func getSymbolScope(symbol *ast.Symbol) *ast.Node {
+	// If this is the symbol of a named function expression or named class expression,
+	// then named references are limited to its own scope.
+	valueDeclaration := symbol.ValueDeclaration
+	if valueDeclaration != nil && (valueDeclaration.Kind == ast.KindFunctionExpression || valueDeclaration.Kind == ast.KindClassExpression) {
+		return valueDeclaration
+	}
+
+	if len(symbol.Declarations) == 0 {
+		return nil
+	}
+
+	declarations := symbol.Declarations
+	// If this is private property or method, the scope is the containing class
+	if symbol.Flags&(ast.SymbolFlagsProperty|ast.SymbolFlagsMethod) != 0 {
+		privateDeclaration := core.Find(declarations, func(d *ast.Node) bool {
+			return checker.HasModifier(d, ast.ModifierFlagsPrivate) || ast.IsPrivateIdentifierClassElementDeclaration(d)
```

Comment:
aside--if this port didn't skip any Strada jsdoc checks for `@private`, then Corsa's going to be more a lot more accurate for JS, since my next PR is to convert `@private` into ModifierFlagsPrivate. (Nothing for you to do here, just an observation.)